src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs

```diff
@@ -436,6 +436,15 @@ private static ApiResponseType CreateDefaultApiResponseType(Type responseType)
     private static EndpointModelMetadata CreateModelMetadata(Type type) =>
         new(ModelMetadataIdentity.ForType(type));
 
+    private static EndpointModelMetadata CreateModelMetadata(IParameterBindingMetadata parameter, Type type)
+    {
+        if (parameter.ParameterInfo?.Member is PropertyInfo propertyInfo && propertyInfo.DeclaringType is not null)
+        {
+            return new(ModelMetadataIdentity.ForProperty(propertyInfo, type, propertyInfo.DeclaringType));
+        }
+        return CreateModelMetadata(type);
```

Comment:
So we can capture parameter info even if the argument is not `AsParameters`. This might be helpful for other scenarios where the model metadata needs to be more complete.

```suggestion
        return ModelMetadataIdentity.ForParameter(parameter);
```